crates/ruff_python_parser/src/string.rs

```diff
@@ -230,6 +230,7 @@ impl<'a> StringParser<'a> {
                 // match a python 3.8 self documenting expression
                 // format '{' PYTHON_EXPRESSION '=' FORMAT_SPECIFIER? '}'
                 '=' if self.peek() != Some('=') && delimiters.is_empty() => {
+                    trailing_seq.push('=');
```

Comment:
I like it that we now use the text ranges to get `leading` and `before_eq`. Would it be possible also to use text ranges to extract `trailing`? It would reduce 2 string allocations for each self-documenting string (one for `trailing_seq`, and one for formatting `trailing` in debug text.

Reply:
i think so. we'd need to store the `=` and any trailing spaces inside `expression` which is a bit odd but doable if we just track where the actual expression ends. will push something with what i mean (and maybe you have ideas for how to improve that)